src/core/container.rs

```diff
@@ -162,6 +134,32 @@ where
         .unwrap_or_else(|_| panic!("container {} has missing or invalid bridge IP", self.id))
     }
 
+    pub fn exec(&self, exec_commands: Vec<ExecCommand>) {
```

Comment:
I think this public API should just take a single `ExecCommand`.

For the `Image` it makes sense to return a list but any caller can run this in a loop themselves.